.taskcluster.yml

```diff
@@ -5,23 +5,112 @@ version: 1
 policy:
   pullRequests: public
 tasks:
-  - schedulerId: taskcluster-github
-    created: {$fromNow: ''}
-    deadline: {$fromNow: '2 hours'}
-    expires: {$fromNow: '1 year'}
-    provisionerId: aws-provisioner-v1
-    workerType: github-worker
-    retries: 5
-    payload:
-      maxRunTime: 600
-      image: python:3.7-slim
-      command:
-        - /bin/bash
-        - --login
-        - -cx
-        - exit 0
-    metadata:
-      owner: JohanLorenzo@users.noreply.github.com
-      source: https://github.com/mozilla-releng/mozilla-version/pull/65
-      name: 'mozilla-version - noop'
-      description: 'noop task while https://github.com/mozilla-releng/mozilla-version/pull/64 lands'
+  $let:
+    expires_in: {$fromNow: '1 year'}
+    user: ${event.sender.login}
+
+    # We define the following variable at the very top, because they are used in the
+    # default definition
+    head_branch:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.ref}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.ref}
+        else: ${event.release.target_commitish}
+
+    head_rev:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.sha}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.after}
+        else: ${event.release.tag_name}
+
+    repository:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.repo.html_url}
+      else: ${event.repository.html_url}
+
+    github_repository_full_name:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.base.repo.full_name}
+      else: ${event.repository.full_name}
+
+  in:
+    $let:
+      short_head_branch:
+        $if: 'head_branch[:10] == "refs/tags/"'
+        then: {$eval: 'head_branch[10:]'}
+        else:
+          $if: 'head_branch[:11] == "refs/heads/"'
+          then: {$eval: 'head_branch[11:]'}
+          else: ${head_branch}
+
+      assume_scope_prefix: assume:repo:github.com/${github_repository_full_name}
+    in:
+      $let:
+        default_task_definition:
+          schedulerId: taskcluster-github
+          created: {$fromNow: ''}
+          deadline: {$fromNow: '2 hours'}
+          expires: ${expires_in}
+          provisionerId: aws-provisioner-v1
+          workerType: github-worker
+          retries: 5
+          payload:
+            maxRunTime: 600   # Decision should remain fast enough to schedule a handful of tasks
+            image: mozillamobile/fenix:1.4
```

Comment:
probably it's better to remove `image` completely, it's a bit misleading to see `fenix` here.

Reply:
Oh right, bad copy and paste!